packages/material-ui/src/styles/deprecated_createMuiTheme.d.ts

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { BreakpointsOptions } from './createBreakpoints';
+import { MixinsOptions } from './createMixins';
+import { Palette, PaletteOptions } from './createPalette';
+import { TypographyOptions } from './createTypography';
+import { Shadows } from './shadows';
+import { ShapeOptions } from './shape';
+import { SpacingOptions } from './createSpacing';
+import { TransitionsOptions } from './transitions';
+import { ZIndexOptions } from './zIndex';
+import { Overrides } from './overrides';
+import { Variants } from './variants';
+import { ComponentsProps } from './props';
+import { Theme } from './createMuiTheme';
+
+export type Direction = 'ltr' | 'rtl';
+
+export interface DeprecatedThemeOptions {
+  shape?: ShapeOptions;
+  breakpoints?: BreakpointsOptions;
+  direction?: Direction;
+  mixins?: MixinsOptions;
+  overrides?: Overrides;
+  palette?: PaletteOptions;
+  props?: ComponentsProps;
+  shadows?: Shadows;
+  spacing?: SpacingOptions;
+  transitions?: TransitionsOptions;
+  typography?: TypographyOptions | ((palette: Palette) => TypographyOptions);
+  variants?: Variants;
+  zIndex?: ZIndexOptions;
+  unstable_strictMode?: boolean;
+}
+
+/**
+ * Generate a theme base on the options received.
+ *
+ * @param options Takes an incomplete theme object and adds the missing parts.
+ * @param args Deep merge the arguments with the about to be returned theme.
+ * @returns A complete, ready to use theme object.
+ */
+export default function deprecated_createMuiTheme(
```

Comment:
I wonder about the name of the helper. Maybe we should call it `createV4MuiTheme()` and have it output a deprecation warning in the console. using v4 in the name can make it more obvious that it's meant to adapt with the theme structure of v4.

Alternatively, I wonder about this alternative approach:

```jsx
import { createMuiTheme, adaptV4Theme } from '@material-ui/core';

const theme = createMuiTheme(adaptV4Theme({}));
```

Reply:
Yep, this looks better, I even have a function for it already 👍 will refactor it

Reply:
Done